libraries/AP_Scripting/examples/plane-doublets.lua

```diff
@@ -48,109 +46,78 @@ function retry_set_mode(mode)
 end
 
 function doublet()
+    local now = millis()
+    local pre_doublet_mode = vehicle:get_mode()
+    local pre_doublet_elevator = SRV_Channels:get_output_pwm(K_ELEVATOR)
+    local pre_doublet_aileron = SRV_Channels:get_output_pwm(K_AILERON)
+    local pre_doublet_rudder = SRV_Channels:get_output_pwm(K_RUDDER)
+    local pre_doublet_throttle = 1200
+    local doublet_choice_pwm = rc:get_pwm(DOUBLET_CHOICE_CHANNEL)
     local callback_time = 100
-    if arming:is_armed() == true and rc:get_pwm(DOUBLET_ACTION_CHANNEL) > 1700 and end_time ==-1 then
-        -- try for the best timing possible while performing a doublet
-        callback_time = DOUBLET_TIME / 10
-        -- start a quick doublet based on some math/logic
-        now = millis()
+    if arming:is_armed() == true and rc:get_pwm(DOUBLET_ACTION_CHANNEL) > 1700 then
```

Comment:
You could use `rc:get_aux_cached(300) == 2`. 300 is the scripting rc input, the user can then assign it to whatever `RCx_OPTION they like. It also allows external trigger via GCS.